mlrun/datastore/datastore_profile.py

```diff
@@ -283,7 +308,10 @@ def url(self, subpath) -> str:
         if subpath.startswith("/"):
             #  in gcs the path after schema is starts with bucket, wherefore it should not start with "/".
             subpath = subpath[1:]
-        return f"gcs://{subpath}"
+        if self.bucket:
+            return f"gcs://{self.bucket}/{subpath}"
+        else:
+            return f"gcs://{subpath}"
```

Comment:
Looks like this is an existing issue, but there's something wrong with the subpath being assumed to not start with a slash here, but the opposite assumption is made in S3.

Reply:
Like you said - its a separate issue.

Reply:
I guess at least add a TODO?

Reply:
I guess in S3 though.